src/ir_Trotec.cpp

```diff
@@ -491,21 +494,32 @@ uint8_t IRTrotec3550::getMode(void) const { return _.Mode; }
 /// Set the temperature.
 /// @param[in] degrees The temperature in degrees.
 /// @param[in] celsius Use celsius units. True, Celsius; False Fahrenheit.
-/// @note Fahrenheit is not yet supported. Waiting on data analysis.
 void IRTrotec3550::setTemp(const uint8_t degrees, const bool celsius) {
   setTempUnit(celsius);
   uint8_t minTemp = kTrotec3550MinTempC;
   uint8_t maxTemp = kTrotec3550MaxTempC;
+  if (!celsius) {  // Fahrenheit?
+    minTemp = kTrotec3550MinTempF;
+    maxTemp = kTrotec3550MaxTempF;
+  }
   uint8_t temp = std::max(degrees, minTemp);
   temp = std::min(temp, maxTemp);
-  _.Temp = _.Temp2 = temp - minTemp;
+  if (celsius) {
+    _.TempC = temp - minTemp;
+    _.TempF = celsiusToFahrenheit(temp) - kTrotec3550MinTempF;
+  } else {
+    _.TempF = temp - minTemp;
+    _.TempC = fahrenheitToCelsius(temp) - kTrotec3550MinTempC;
+  }
 }
 
 /// Get the current temperature setting.
 /// @return The current setting for temp. in degrees.
-/// @note Fahrenheit is not yet supported. Waiting on data analysis.
 uint8_t IRTrotec3550::getTemp(void) const {
-  return _.Temp + kTrotec3550MinTempC;
+  if (getTempUnit())  // Is Celsius?
+    return _.TempC + kTrotec3550MinTempC;
+  else
+    return _.TempF + kTrotec3550MinTempF;
```

Comment:
This is more of a question, since there is a return the `else` is not needed, would it be better to skip it.
and next part would be to use `return getTempUnit() ? _.TempC + kTrotec3550MinTempC : _.TempF + kTrotec3550MinTempF;` instead ?

Reply:
Done.

Funnily enough, I did think of all of those options and thought, that version was more readable and that the compiler will _probably_ optimise it down to the same thing, and I was being lazy! ;-)

I've gone with your single line solution to be more consistent in the code base.